### PR TITLE
Add a doc for the command line tool

### PR DESCRIPTION
**Pull request by yang-g**

### CODE REVIEW

**Comment by yang-g** [comment]
This is a doc only change.

File: doc/command_line_tool.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+## Overview
+
+This document describes the command line tool that comes with gRPC repository. It is desireable to have command line
+tools written in other languages to roughly follow the same syntax and flags.
+
+At this point, the tool needs to be built from source, and it should be moved out to grpc-tools repository as a stand
+alone application once it is mature enough.
+
+## Core funcitonality
```
Comment (vjpai) [inline doc/command_line_tool.md]: functionality

File: doc/command_line_tool.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+The command line tool should support the following things:
+
+- List server services and methods through server reflection.
+- Infer request/response types from server reflection result.
+- Fine-grained auth control (such as, use this oauth token to talk to the server).
+- Send streaming rpc.
+
+## Code location
+
+To use the tool, you need to get the grpc repository and in the grpc directory call
```
Comment (vjpai) [inline doc/command_line_tool.md]: call -> execute?

File: doc/command_line_tool.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+```
+make grpc_cli
```
Comment (vjpai) [inline doc/command_line_tool.md]: Add a shell prompt?

File: doc/command_line_tool.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+## Usage
+
+### Basic usage
+
+Send a rpc to a helloworld server at `localhost:50051`:
+
+```
+bins/opt/grpc_cli call localhost:50051 SayHello examples/protos/helloworld.proto \
+    "name: 'world'"  --enable_ssl=false
```
Comment (vjpai) [inline doc/command_line_tool.md]: shell prompt?

File: doc/command_line_tool.md (hunk 1 above)
Comment (yang-g) [inline doc/command_line_tool.md]: Done.

File: doc/command_line_tool.md (hunk 2 above)
Comment (yang-g) [inline doc/command_line_tool.md]: Done.

File: doc/command_line_tool.md (hunk 3 above)
Comment (yang-g) [inline doc/command_line_tool.md]: Done.

File: doc/command_line_tool.md (hunk 4 above)
Comment (yang-g) [inline doc/command_line_tool.md]: Done.

**Comment by vjpai** [comment]
LGTM

**Pull request merged**